extensions/interactions/ItemSelectionInput/directives/ItemSelectionInputValidationService.js

```diff
@@ -148,6 +148,35 @@ oppia.factory('ItemSelectionInputValidationService', [
             return handledAnswer;
           });
         }
+        var answerChoiceToIndex = {};
+        seenChoices.forEach(function(seenChoice, choiceIndex) {
+          answerChoiceToIndex[seenChoice] = choiceIndex;
+        });
+
+        answerGroups.forEach(function(answerGroup, answerIndex) {
+          var rules = answerGroup.rules;
+          rules.forEach(function(rule, ruleIndex) {
+            var ruleInputs = rule.inputs.x;
+            ruleInputs.forEach(function(ruleInput) {
+              var choiceIndex = answerChoiceToIndex[ruleInput];
+              if (rule.type === 'IsProperSubsetOf') {
+                if (ruleInputs.length < 2) {
+                  warningsList.push({
+                    type: WARNING_TYPES.ERROR,
+                    message: (
+                      'In answer group ' + (answerIndex + 1) + ', ' +
+                      'rule ' + (ruleIndex + 1) + ', the "proper subset" ' +
+                      'rule must include at least 2 options.')
+                  });
+                }
+                handledAnswers[choiceIndex] = true;
+              }
+            });
+          });
+        });
+        areAllChoicesCovered = handledAnswers.every(function(handledAnswer) {
```

Comment:
Are you sure this is correct? You've only considered one type of rule (for the != 1 case). I am not sure you need this actually.

Reply:
HI! @seanlip,
 This rule was considered for all cases, Also The current validation does not consider cases where rule type = "equals", and mininputs>answergroup.length,  And when rule type is "contains at least one of", and answergroup.length (number of choices chosen) is 0, both put user into an infinite loop.  I feel we can address this in a different issue. And for the All choices covered, I feel it is needed. (tried shifting it up (below the if statement) and running oppia) .

Reply:
Also sorry for late replies again, This will not happen after 16th (sem officially gets over, (ill have end sem exams but most of those are cramming night before)), I will be much more active frm 16th.
Thanks!

Reply:
What I'm trying to say is, I'm not sure this check works correctly for the != 1 case. (That's why it was originally only implemented for the == 1 case.)

That's because you're populating handledAnswers only using IsProperSubsetOf and nothing else.

Reply:
Yeah this was another problem, I am not sure how is handled ( I tried checking what is happening in OutcomeObjectFactory,) and how we are checking if all options are handled.  The handled choices is used to see if some choice is not handled in the ==1 case, what I am doing is adding all the other choices, but verifying that all are checked (if they have proper subset rule). I think we can solve this in #6418 , by adding other options and validating them. and that way we very each choice is addressed. 
Note: yeah all cases are handled but not in a proper way, here if all choices are 1 then we verify it here, If all choices are not 1, then that variable is anyway false.

Reply:
Also, I think I got the question wrong, Yeah the check works for !=1 case, as  areAllChoicesCovered will only be true if All choices have been checked by some rule.

Reply:
I don't understand the logic here. In the maxAllowedCount > 1 case, handledAnswers is initialized [here](https://github.com/oppia/oppia/pull/6350/files#diff-6ceab13d6821a6546bcd8f966a413944R104), the entire block from lines 111-147 is skipped (since maxAllowedCount > 1), and then it seems to me like all rule types apart from IsProperSubsetOf are being ignored.

So, why do you say that "areAllChoicesCovered will only be true if all choices have been checked by some rule"? There are other rules besides isProperSubsetOf. I recommend that you do not try to extend that check beyond the `== 1` case in this PR.

Reply:
Well Yeah I agree, I am confused now,  I will remove that for !=1 case, 
And this is what I was thinking, when I put it in, 
I modified the ItemSelectionValidationService code to this, 
```javascript
if (!areAllChoicesCovered) {     // Now, this will happen if all choices are not covered by =    == 1 case or properSubset rule, I am checking this by not including "selection3"
179 //          if (!defaultOutcome || defaultOutcome.isConfusing(stateName)) {  //this I am guessing che    cks if user has given default feedback (if the user has given oppia something to say, I tried backtracking this, so now Im assuming user has not given a def feedback, So I shuld get this error
180             warningsList.push({
181               type: WARNING_TYPES.ERROR,
182               message: (
183                 'Please add something for Oppia to say in the ' +
184                 '\"All other answers\" response.')
185             });
186 //          }
187         }
```
and added this check in the Spec file,  
```javascript
 92   it('If everything goes how I except, this case should be handled.', function() {
 93     customizationArguments.maxAllowableSelectionCount.value=1;  //this is the case handled by ===1 ca    se, 
 94     var warnings = validatorService.getAllWarnings(currentState, customizationArguments, IsProperSubs    etValidOption,goodDefaultOutcome);
 95     console.log(warnings);
 96   });
 97 
```
And checked when options cover everything and when they do not (adding and removing option 3 and checking for warnings) . But yeah this was getting confusing, So I think It is better I drop that, and check for in the ===1 for ProperSubsetRule also case, and make another !=1 case for the proper subset rule. 
Ill change it now and commit.